openhtf/exe/__init__.py

```diff
@@ -209,3 +215,7 @@ def _ExecuteTestPhases(self, executor):
         break
     else:
       self._test_state.SetStateFinished()
+    # Run teardown function.
+    if self._teardown_function:
+      self._test_state.pending_phases.insert(0, self._teardown_function)
```

Comment:
I would argue there's no need to add this to pending phases if we don't think of it as a phase.